eval/eval-scripts/mmlu.py

```diff
@@ -0,0 +1,138 @@
+
+    # Check if all required files are present
+    if all(file in existing_files for file in required_files):
+        logger.debug(f"Model directory {local_model_path} contains required files. Skipping download.")
+    else:
+        os.makedirs(local_model_path, exist_ok=True)
+        logger.debug(f"Downloading model to {local_model_path} from {model_s3_path}")
+
+        try:
+            response = s3_client.list_objects_v2(Bucket=bucket_name, Prefix=model_s3_path)
+            if 'Contents' not in response:
+                logger.debug(f"No objects found at {model_s3_path}. Check the S3 path and permissions.")
+                return
+
+            for obj in response['Contents']:
+                key = obj['Key']
+                local_file_path = os.path.join(local_model_path, os.path.basename(key))
+                s3_client.download_file(bucket_name, key, local_file_path)
+                logger.debug(f"Downloaded {key} to {local_file_path}")
+
+            # Verify that the expected files are downloaded
+            existing_files = os.listdir(local_model_path)
+            if all(file in existing_files for file in required_files):
+                logger.debug(f"Required model files found at {local_model_path}.")
+            else:
+                logger.debug(f"Model files not found at {local_model_path}. Expected files: {required_files}. Check the S3 path and permissions.")
+
+        except Exception as e:
+            logger.debug(f"Error downloading model from S3: {e}")
+            raise
+
+    # Debug: Check GPU availability
+    gpu_available = torch.cuda.is_available()
+    gpu_count = torch.cuda.device_count()
+    current_device = torch.cuda.current_device() if gpu_available else None
+    gpu_name = torch.cuda.get_device_name(current_device) if gpu_available else "No GPU available"
+
+    logger.debug(f"GPU Available: {gpu_available}")
+    logger.debug(f"Number of GPUs: {gpu_count}")
+    if gpu_available:
+        logger.debug(f"Using GPU: {current_device} - {gpu_name}")
+    else:
+        logger.debug("Falling back to CPU.")
+
+    device = device or ("cuda" if gpu_available else "cpu")
+
+    logger.debug("Running MMLU evaluation...")
+    evaluator = MMLUEvaluator(
+        model_path=local_model_path,
+        tasks=mmlu_tasks,
+        model_dtype=model_dtype,
+        few_shots=few_shots,
+        batch_size=batch_size,
+        device=device,
+    )
+
+    overall_score, _ = evaluator.run()
+
+    logger.debug("MMLU eval phase -- Clearing PyTorch cache")
+    torch.cuda.empty_cache()
+
+    with open(output_file, 'w') as f:
+        f.write(f"Overall MMLU Score: {overall_score}")
+    logger.debug(f"Saved evaluation results to {output_file}")
+
+    try:
+        output_prefix = '/'.join(output_s3_path.split('/')[3:])
+        s3_client.upload_file(output_file, bucket_name, f"{output_prefix}/{os.path.basename(output_file)}")
```

Comment:
You can produce `dsl.Output` instead of uploading to S3 directly.

If you need output artifact path - local to the pod (you can write to it as if it was a file/folder), you can use:

```python
a: dsl.Output

with open(a.path, "w") as f:
...
```

Reply:
yup

> WIP: Don't Review!! Don't try to run (probably won't run anyways) :)